zerver/tests/test_upload.py

```diff
@@ -1019,13 +1031,18 @@ def test_delete_avatar(self) -> None:
         """
         A DELETE request to /json/users/me/avatar should delete the profile picture and return gravatar URL
         """
-        self.login(self.example_email("hamlet"))
-        hamlet = self.example_user('hamlet')
-        hamlet.avatar_source = UserProfile.AVATAR_FROM_USER
-        hamlet.save()
+        self.login(self.example_email("cordelia"))
+        cordelia = self.example_user("cordelia")
+        cordelia.avatar_source = UserProfile.AVATAR_FROM_USER
+        cordelia.save()
 
+        do_set_realm_property(cordelia.realm, 'avatar_changes_disabled', True)
         result = self.client_delete("/json/users/me/avatar")
-        user_profile = self.example_user('hamlet')
+        self.assert_json_error(result, "Avatar changes have been disabled by the admin.", 400)
+
+        do_set_realm_property(cordelia.realm, 'avatar_changes_disabled', False)
+        result = self.client_delete("/json/users/me/avatar")
+        user_profile = self.example_user("cordelia")
```

Comment:
Just wondering, why this test does not work (even before the change), if using the `cordelia` variable from above instead of `user_profile`?